File: yak-gl-util/src/main/java/net/yakclient/opengl/util/SlickFontWrapper.java
package net.yakclient.opengl.util;

import org.newdawn.slick.TrueTypeFont;

public class SlickFontWrapper implements YakFont {
    private final java.awt.Font type;
    private final TrueTypeFont trueType;
    private final FontMetaData metaData;

    public SlickFontWrapper(java.awt.Font type, TrueTypeFont trueType, FontMetaData metaData) {
        this.type = type;
        this.trueType = trueType;
        this.metaData = metaData;
    }

    @Override
    public java.awt.Font getType() {
        return this.type;
    }

    @Override
    public String getName() {
        return this.type.getName();
    }

    @Override
    public FontMetaData getMeta() {
        return this.metaData;
    }

    @Override
    public void drawString(String value, RGBColor color, double x, double y) {
        this.trueType.drawString((float) x, (float) y, value, SlickTranslator.toColor(color));
    }

    @Override
    public void drawString(String value, double x, double y) {
        this.drawString(value, ColorCodes.WHITE, x, y);
    }
}

File: util/src/main/kotlin/net/yakclient/graphics/util/SlickFontWrapper.kt
package net.yakclient.graphics.util

//public class SlickFontWrapper(
//    override val type: Font,
//    private val trueType: TrueTypeFont,
//    override val meta: FontMetaData,
//) : YakFont {
//    override val name: String
//        get() = type.name
//
//    override val lineHeight: Double
//        get() = trueType.lineHeight.toDouble()
//
//
//    override fun getWidth(value: String): Int =
//        trueType.getWidth(value)
//
//
//    override fun getHeight(value: String): Int =
//        trueType.getHeight(value)
//
//
//    override fun stringContext(value: String, color: Color, x: Double, y: Double) {
//        trueType.drawString(x.toFloat(), y.toFloat(), value, SlickTranslator.toColor(color))
//    }
//
//    override fun stringContext(value: String, x: Double, y: Double) {
//        this.stringContext(value, ColorCodes.WHITE, x, y)
//    }
//}